### PR TITLE
Redux configuration

### PR DESCRIPTION
**Pull request by krisbobo**
## Milestone 2: Add reducers and actions
**The following changes are being introduced:**
- Introducing redux toolkit in the project
- Configured a redux store
- Defined a sliced of state for books and categories

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    case ADD_BOOK:
+      return [
+        ...state,
+        {
+          id: idCounter + 1,
+          title: action.title,
+          author: action.author,
+        }];
+    case REMOVE_BOOK:
+      return remove(state, action.id);
```
Comment (Deepakdanger) [inline src/redux/books/books.js:21]: - [ ] [OPTIONAL] Here kindly use directly the above function`state.filter((book) => book.id !== bookID);` here only.

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  state.filter((book) => book.id !== bookID);
+};
+
+export default (state = initialState, action = {}) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      return [
+        ...state,
+        {
+          id: idCounter + 1,
```
Comment (Deepakdanger) [inline src/redux/books/books.js:16]: - [ ] Instead of creating id like this use `Math.floor(Math.random() * 100)`.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @krisbobo , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 2 above)
Comment (krisbobo) [reply src/redux/books/books.js:16]: Hi @Deepakdanger
I don't understand why you want me to implement these changes as you haven't attached any reason as to why the changes are necessary.

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @krisbobo, 

To highlight 👍🏽 
✔️ App runs smoothly
✔️ No errors
✔️ README is professional

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
- Kindly note that generating IDs using ```Math.floor(Math.random() * 100)``` ensures that each ID is unique, as the chance of two randomly generated numbers being the same is very low. In contrast, using a counter means that if the counter is not properly managed or reset, it could potentially generate duplicate IDs.



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Comment by omarsalem7** [comment]
Hi @krisbobo 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**